crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -335,38 +335,37 @@ fn handle_in_between_bodies_end_of_line_comment<'a>(
             return CommentPlacement::Default(comment);
         }
 
-        if !locator.contains_line_break(TextRange::new(preceding.end(), comment.slice().start())) {
-            // Trailing comment of the preceding statement
+        if locator.contains_line_break(TextRange::new(preceding.end(), comment.slice().start())) {
+            // The `elif` or except handlers have their own body to which we can attach the trailing comment
             // ```python
-            // while test:
-            //     a # comment
-            // else:
+            // if test:
+            //     a
+            // elif c: # comment
             //     b
             // ```
-            if preceding.is_node_with_body() {
-                // We can't set this as a trailing comment of the function declaration because it
-                // will then move behind the function block instead of sticking with the pass
+            if following.is_except_handler() {
+                return CommentPlacement::trailing(following, comment);
+            } else if following.is_stmt_if() {
+                // We have to exclude for following if statements that are not elif by checking the
+                // indentation
                 // ```python
                 // if True:
-                //     def f():
-                //         pass  # a
-                // else:
+                //     pass
+                // else:  # Comment
+                //     if False:
+                //         pass
                 //     pass
                 // ```
-                CommentPlacement::Default(comment)
-            } else {
-                CommentPlacement::trailing(preceding, comment)
+                let base_if_indent =
```

Comment:
Just an idea. It may be annoying to do because of all the type narrowing that's necessary and you would need to identify the right parent body but I thought it may be worth mentioning. The way to normally detect whether it is an `elif` or not... oh no.... It is not. These two programs have the same AST

```python
if True:
    pass
elif False:
    pass
```

```
If(
    StmtIf {
        range: 0..38,
        test: Constant(
            ExprConstant {
                range: 3..7,
                value: Bool(
                    true,
                ),
                kind: None,
            },
        ),
        body: [
            Pass(
                StmtPass {
                    range: 13..17,
                },
            ),
        ],
        orelse: [
            If(
                StmtIf {
                    range: 18..38,
                    test: Constant(
                        ExprConstant {
                            range: 23..28,
                            value: Bool(
                                false,
                            ),
                            kind: None,
                        },
                    ),
                    body: [
                        Pass(
                            StmtPass {
                                range: 34..38,
                            },
                        ),
                    ],
                    orelse: [],
                },
            ),
        ],
    },
),
```

```python
if True:
    pass
else:
    if False:
        pass
```

```
If(
        StmtIf {
            range: 41..91,
            test: Constant(
                ExprConstant {
                    range: 44..48,
                    value: Bool(
                        true,
                    ),
                    kind: None,
                },
            ),
            body: [
                Pass(
                    StmtPass {
                        range: 54..58,
                    },
                ),
            ],
            orelse: [
                If(
                    StmtIf {
                        range: 69..91,
                        test: Constant(
                            ExprConstant {
                                range: 72..77,
                                value: Bool(
                                    false,
                                ),
                                kind: None,
                            },
                        ),
                        body: [
                            Pass(
                                StmtPass {
                                    range: 87..91,
                                },
                            ),
                        ],
                        orelse: [],
                    },
                ),
            ],
        },
    ),
```